### ISSUE TITLE
Condition-specific cell types - Can you run multinichenetr on datasets that have varying cell type composition?

### ISSUE THREAD

**Issue by olympiahardy** [saeyslab/multinichenetr#32]
Hi there, 

Thank you for all the hard work on developing this tool! 

I was wondering if/how multinichenetr can be run on a dataset that has some cell types that are condition specific. For example if I had a macrophage sub-population that was present in one condition but completely absent in the other I wouldn't be able to run the differential expression analysis. I know that CellChat has a functionality where you can harmonise the cell type annotations across conditions by setting the expression values to 0 where it is absent in a given condition, but then that matters less when the tool is only doing a comparative analysis rather than a differential expression for ligand receptor pairs between conditions. 

I hope that all makes sense, I'd love to hear your thoughts or perhaps suggestions to workaround this. At the moment all I have is to try and harmonise the cell type labels by using broader annotations but then you lose the granularity of sub-clusters. 

Thanks!

Olympia

**Comment by browaeysrobin** [comment]
Hi @olympiahardy 

In the current release, there is indeed no possibility to handle condition-specific cell types. It has been on our mind to add this option in a future release.

For the moment you can indeed use a broader annotation for that specific cell type. Some ligands/receptors/target genes in this output may be reflective of the condition-specific subpopulation.

**Comment by Boehmin** [comment]
Hi @browaeysrobin,
I am also working with the same issue. You mentioned the following in the closed issue [#38 ](https://github.com/saeyslab/multinichenetr/issues/38)
`Related to this, you can check issue https://github.com/saeyslab/multinichenetr/issues/32 and the solution I propose there; we will be releasing a work-around to handle condition-specific cell types soon (within one month).`

Has this work-around been released? I have those clusters "somewhat" annotated in all my conditions, but one for example has only 5 cells in one of my conditions. This does not necessarily inspire confidence for DEG. I`m not sure how else to approach this, or what else is referred to by "broader" annotation. We have single nuclei data, this cluster of nuclei is virtually non-existent in a healthy condition so I am not sure how a "broader" annotation would fix this, or what to annotate these nuclei as.

**Comment by browaeysrobin** [comment]
Hi @Boehmin 

You can find the current workaround for this by:
1) installing the dev-branch of this package (`devtools::install_github("saeyslab/multinichenetr", "dev-branch")`)
2) following the vignette: https://github.com/saeyslab/multinichenetr/blob/dev-branch/vignettes/condition_specific_celltype_MISC.Rmd

If steps in this workflow don't work anymore, let me know.

**Comment by olympiahardy** [comment]
Hi @browaeysrobin,

Thank you for releasing this vignette, I have tested it on my own data and haven't run into any problems! I have a question regarding interpretation as I want to ensure I'm not drawing any false conclusions. 

If we compare two circos plots in my condition of interest (M) compared to (A/S):

- **Circos X** - This being generated in the regular workflow where all cell types are present in the conditions and the interactions have been prioritised by the usual multinichenetr fashion
- **Circos Y** - This being generated after running Workflow A/B (condition-specific cells as senders/receivers) and visualising the combined prioritisation table at the end of the new vignette

Circos X shows the top 50 prioritised predicted ligand/receptors for cell types present in this condition found to be differentially expressed in condition M when compared to conditions A and S. These prioritised ligand/receptor interactions can be discussed in relation to all three conditions in the experiment.

Circos Y shows the top 50 prioritised predicted ligand/receptors for all cell types in the dataset with the inclusion of condition-specific cell types. Where this is true, for these cell types the predicted ligand/receptors have been prioritised based on gene expression within condition M rather than differential expression across conditions. The returned ligand/receptor interactions between condition-specific cell types may reflect the underlying biology specific to condition M but are not directly comparable to the other conditions in this analysis. 

And then regarding other visualisations such as the ligand activity plot, if I adapted the source code to take the dataframe  _multinichenet_output$prioritization_tables_with_condition_specific_celltype_receiver$ligand_activities_target_de_tbl_ instead of the list _ligand_activities_targets_DEgenes_ am I correct in understanding that I could then visualise downstream targets associated with these condition-specific cell types? To clarify if not clear, does the ligand_activities_target_de_tbl in condition-specific analysis equate to ligand_activities_targets_DEgenes in the usual multinichenetr analysis? 

Sorry for the long message and thanks again for the vignette!
